lib/tests/streamlit/external/langchain/streamlit_callback_handler_test.py

```diff
@@ -16,17 +16,28 @@
 
 import unittest
 from pathlib import Path
+from unittest.mock import MagicMock
 
-import langchain
 import pytest
 import semver
 from google.protobuf.json_format import MessageToDict
 
 import streamlit as st
 from tests.delta_generator_test_case import DeltaGeneratorTestCase
-from tests.streamlit.external.langchain.capturing_callback_handler import (
-    playback_callbacks,
-)
+
+try:
+    import langchain
+
+    from tests.streamlit.external.langchain.capturing_callback_handler import (
+        playback_callbacks,
+    )
+except ImportError:
+    # If langchain isn't installed (it doesn't work with Python 3.12 yet),
+    # we'll skip this test.
+    # TODO[kajarenc] Remove this once langchain supports Python 3.12.
+    langchain = MagicMock(__version__="0.0.0")
+    playback_callbacks = None
+    pytestmark = pytest.mark.skip(reason="Langchain doesn't support Python 3.12 yet.")
```

Comment:
nit: do we have to bind the return value of `pytest.mark.skip` to a variable here?

Reply:
@vdonato we need this assignment so skip all tests of this file in case if langchain does not installed: https://doc.pytest.org/en/latest/how-to/skipping.html#skip-all-test-functions-of-a-class-or-module